### ISSUE TITLE
How much data is sufficient for transformer tts training?

### ISSUE THREAD

**Issue by enamoria** [espnet/espnet#1245]
Hello, I can't train a transformer system that can synthesize audible speech. I only have 6 hours of speech. I've successfully trained tacotron2 (with nvidia implementation, not with esp) so I think 6 hours are quite sufficient, but I can't make any audible audio with transformer. 
I've tried 1000 epochs, no good result, with single GPU, small batch but accum_grad=8

I think the reason is i have only 6 hours of data and the number of character in the training set is up to 100, which make the model harder to learning something

Any advice? Thanks very much

Here is some result from att_ws. I dont have loss log (I accidentally delete it)
[att_ws.zip](https://github.com/espnet/espnet/files/3701531/att_ws.zip)

**Comment by Pydataman** [comment]
@enamoria  English corpus?
I have trained tacotron、tacotron2、transformer tts and dc_tts on chinese corpus.it has about 6 hours.
The effect of dc_tts model is better than others.For transformer tts, 6 hours of data is quite small.

**Comment by enamoria** [comment]
> @enamoria English corpus?
> I have trained tacotron、tacotron2、transformer tts and dc_tts on chinese corpus.it has about 6 hours.
> The effect of dc_tts model is better than others.For transformer tts, 6 hours of data is quite small.

But are you able to get audible speech with transformer? I can't even get intelligible speech

**Comment by kan-bayashi** [comment]
Hi @enamoria.
I check your figures. It seems that the utterances contains long silence in the back and forth.
![8 after_postnet_fbank ep 1000](https://user-images.githubusercontent.com/22779813/66450430-78d7a000-ea93-11e9-95c0-bdb61a7612be.png)
Trimming these parts helps to learn attention.
You can check this code to perform silence trimming.
https://github.com/espnet/espnet/blob/master/egs/m_ailabs/tts1/local/trim_silence.sh

**Comment by kan-bayashi** [comment]
And we can successfully trained Transformer using Japanese Kana or Phoneme with 10 hours data.
You can check the results.
https://github.com/espnet/espnet/blob/master/egs/jsut/tts1/RESULTS.md

My recommendation is that:
- Trim the silence (Might need to tune some threshold)
- Try a small transformer with reduction factor (v2) and check it works

**Comment by Pydataman** [comment]
@enamoria no

**Comment by Pydataman** [comment]
@kan-bayashi can you try chinese tts use espnet? I do‘t know how to deal chinese corpus label，use pinyin or phoneme?

**Comment by kan-bayashi** [comment]
@Pydataman I will make the recipe using https://www.data-baker.com/open_source.html

**Comment by Pydataman** [comment]
@kan-bayashi thanks。I am training transformer tts model use own chinese corpus.

**Comment by kan-bayashi** [comment]
I finished making the recipe #1259.
The training is not yet finished but it seems working well with transformer.

**Comment by Pydataman** [comment]
@kan-bayashi thanks

**Comment by kan-bayashi** [comment]
@Pydataman I generated samples using Transformer @ 200 epochs.
How do you think this samples?
I cannot understand the meaning but the quality sounds nice.
https://drive.google.com/open?id=1Si5CJ-b3iLKjW5bs-0Ll2EBnoCqMO_3n

**Comment by lingjzhu** [comment]
@kan-bayashi Just FYI, this sample sounds good and the sentence makes perfect sense to me.

**Comment by Pydataman** [comment]
@kan-bayashi sounds good, very clear

**Comment by enamoria** [comment]
> And we successfully trained Transformer using Japanese Kana or Phoneme with 10 hours data.
> You can check the results.
> https://github.com/espnet/espnet/blob/master/egs/jsut/tts1/RESULTS.md
> So I think it can be trained with 6 hours data (but might has some problems on pronunciation).
> 
> My recommendation is that:
> 
> * Trim the silence (Might need to tune some threshold)
> * Check the text is correctly normalized and cleaned
> * Try a small transformer with reduction factor (v2) and check it works

@kan-bayashi Your recommendation helped. With some little modification in processing data, I'm able to synthesize better speech. Thanks for your suggestion.
I can make a PR for Vietnamese recipes, is that ok?

**Comment by kan-bayashi** [comment]
@enamoria That is very coool!
We are really welcome your PR!
